sys/include/net/gnrc/netif.h

```diff
@@ -370,6 +370,25 @@ gnrc_netif_t *gnrc_netif_iter(const gnrc_netif_t *prev);
  */
 gnrc_netif_t *gnrc_netif_get_by_pid(kernel_pid_t pid);
 
+/**
+ * @brief   Get network interface by netdev
+ *
+ * @param[in] dev   A netdev pointer of a network interface.
+ *
+ * @return  The network interface on success.
+ * @return  NULL, if no network interface for the netdev exists.
+ */
+static inline gnrc_netif_t *gnrc_netif_get_by_netdev(const netdev_t *dev)
```

Comment:
Note there's [on-going work](https://github.com/RIOT-OS/RIOT/pull/13614) for removing the `gnrc_netif` <-> `netdev` dependency. Could this be closer to layer that requires this feature?